streams/shared/src/main/scala/zio/stream/internal/ChannelExecutor.scala

```diff
@@ -1005,6 +1005,15 @@ private[zio] object SingleProducerAsyncInput {
       .flatMap(p => Ref.make[State[Err, Elem, Done]](State.Empty(p)))
       .map(new SingleProducerAsyncInput(_))
 
+  object unsafe {
+    def make[Err, Elem, Done](fiberId: FiberId)(implicit trace: Trace): SingleProducerAsyncInput[Err, Elem, Done] = {
```

Comment:
2 minor comments here:
1. To keep things consistent, add an implicit `unsafe: Unsafe` requirement for this method. This way you won't need to pass it explicitly below either 
2. Change the implementation of `SingleProducerAsyncInput.make`  to `ZIO.fiberIdWith(unsafe.make(_)(trace, Unsafe))`

Reply:
Done.

I removed the implict `trace: Trace` from the `unsafe.make` function to make it consistent with `Promise.unsafe.make`, etc.